app/lib/screens/market/buy_tft.dart

```diff
@@ -0,0 +1,625 @@
+import 'package:decimal/decimal.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_keyboard_visibility/flutter_keyboard_visibility.dart';
+import 'package:threebotlogin/helpers/logger.dart';
+import 'package:threebotlogin/helpers/transaction_helpers.dart';
+import 'package:threebotlogin/models/offer.dart';
+import 'package:threebotlogin/models/wallet.dart';
+import 'package:threebotlogin/providers/orders_notifier.dart';
+import 'package:threebotlogin/screens/market/order.dart';
+import 'package:threebotlogin/services/stellar_service.dart' as Stellar;
+import 'package:threebotlogin/widgets/custom_dialog.dart';
+
+class BuyTFTWidget extends StatefulWidget {
+  final Wallet wallet;
+  final Offer? offer;
+  final bool edit;
+  const BuyTFTWidget(
+      {super.key, required this.wallet, this.offer, required this.edit});
+
+  @override
+  State<BuyTFTWidget> createState() => _BuyTFTWidgetState();
+}
+
+class _BuyTFTWidgetState extends State<BuyTFTWidget> {
+  late TextEditingController amountController;
+  late TextEditingController priceController;
+  final totalAmountController = TextEditingController();
+  final FocusNode textFieldFocusNode = FocusNode();
+  String? amountError;
+  String? priceError;
+  bool loading = false;
+  bool loadingPrice = true;
+  double? currentMarketPrice;
+  List percentages = [25, 50, 75, 100];
+  String availableUSDC = '0';
+
+  @override
+  void initState() {
+    super.initState();
+    amountController = TextEditingController(
+        text: widget.edit ? widget.offer?.amount.toString() ?? '' : '');
+    priceController = TextEditingController(
+        text: widget.edit ? widget.offer?.price.toString() ?? '' : '');
+    amountController.addListener(_calculateTotal);
+    priceController.addListener(_calculateTotal);
+    if (widget.edit) _calculateTotal();
+    if (!widget.edit) _fetchCurrentMarketPrice();
+    _getAvailableUSDC();
+  }
+
+  @override
+  void dispose() {
+    textFieldFocusNode.dispose();
+    amountController.dispose();
+    priceController.dispose();
+    totalAmountController.dispose();
+    amountError = null;
+    priceError = null;
+    amountController.removeListener(_calculateTotal);
+    priceController.removeListener(_calculateTotal);
+    super.dispose();
+  }
+
+  _getAvailableUSDC() async {
+    final available = await Stellar.getAvailableUSDCBalance(
+        widget.wallet.stellarSecret, widget.offer);
+    setState(() {
+      availableUSDC = available;
+    });
+  }
+
+  Future<void> _fetchCurrentMarketPrice() async {
+    try {
+      setState(() {
+        loadingPrice = true;
```

Comment:
This loading is defined and updated, but never used. Please use till the price is loaded and set the price to empty in case of failure